// searchx-chat/lib/components/Messages/FileMessage.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _FileIcon = _interopRequireDefault(require("./../icons/FileIcon"));

var _this = void 0,
    _jsxFileName = "/Users/felipemoraes/Developer/react-team-chat-window/src/components/Messages/FileMessage.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var FileMessage = function FileMessage(props) {
  var date = new Date(props.data.date);

  var dateString = /*#__PURE__*/_react["default"].createElement("div", {
    className: "sc-message--date",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 8,
      columnNumber: 5
    }
  }, " ", date.toLocaleTimeString([], {
    timeStyle: 'short'
  }).replace(/(\r\n|\n|\r)/gm, ""), " ");

  return /*#__PURE__*/_react["default"].createElement("a", {
    className: "sc-message--file",
    href: props.data.url,
    download: props.data.fileName,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 11,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react["default"].createElement(_FileIcon["default"], {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 12,
      columnNumber: 7
    }
  }), /*#__PURE__*/_react["default"].createElement("p", {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 13,
      columnNumber: 7
    }
  }, props.data.fileName), dateString);
};

var _default = FileMessage;
exports["default"] = _default;
module.exports = exports.default;
