// searchx-chat/lib/components/ChatWindow.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _MessageList = _interopRequireDefault(require("./MessageList"));

var _UserInput = _interopRequireDefault(require("./UserInput"));

var _Header = _interopRequireDefault(require("./Header"));

var _jsxFileName = "/Users/felipemoraes/Developer/react-team-chat-window/src/components/ChatWindow.js";

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (typeof call === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

var ChatWindow = /*#__PURE__*/function (_Component) {
  _inheritsLoose(ChatWindow, _Component);

  var _super = _createSuper(ChatWindow);

  function ChatWindow(props) {
    return _Component.call(this, props) || this;
  }

  var _proto = ChatWindow.prototype;

  _proto.onUserInputSubmit = function onUserInputSubmit(message) {
    this.props.onUserInputSubmit(message);
  };

  _proto.onFilesSelected = function onFilesSelected(filesList) {
    this.props.onFilesSelected(filesList);
  };

  _proto.render = function render() {
    var messageList = this.props.messageList || [];
    var classList = ['sc-chat-window', this.props.isOpen ? 'opened' : 'closed'];
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: classList.join(' '),
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 28,
        columnNumber: 7
      }
    }, /*#__PURE__*/_react["default"].createElement(_Header["default"], {
      teamName: this.props.agentProfile.teamName,
      imageUrl: this.props.agentProfile.imageUrl,
      onClose: this.props.onClose,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 29,
        columnNumber: 9
      }
    }), /*#__PURE__*/_react["default"].createElement(_MessageList["default"], {
      messages: messageList,
      imageUrl: this.props.agentProfile.imageUrl,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 34,
        columnNumber: 9
      }
    }), /*#__PURE__*/_react["default"].createElement(_UserInput["default"], {
      onSubmit: this.onUserInputSubmit.bind(this),
      onFilesSelected: this.onFilesSelected.bind(this),
      showEmoji: this.props.showEmoji,
      showFile: this.props.showFile,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 38,
        columnNumber: 9
      }
    }));
  };

  return ChatWindow;
}(_react.Component);

ChatWindow.propTypes = process.env.NODE_ENV !== "production" ? {
  agentProfile: _propTypes["default"].object.isRequired,
  isOpen: _propTypes["default"].bool.isRequired,
  onClose: _propTypes["default"].func.isRequired,
  onFilesSelected: _propTypes["default"].func,
  onUserInputSubmit: _propTypes["default"].func.isRequired,
  showEmoji: _propTypes["default"].bool,
  showFile: _propTypes["default"].bool
} : {};
var _default = ChatWindow;
exports["default"] = _default;
module.exports = exports.default;
