// searchx-chat/es/components/Messages/FileMessage.js
var _this = this,
    _jsxFileName = "/Users/felipemoraes/Developer/react-team-chat-window/src/components/Messages/FileMessage.js";

import React from 'react';
import FileIcon from './../icons/FileIcon';

var FileMessage = function FileMessage(props) {
  var date = new Date(props.data.date);
  var dateString = /*#__PURE__*/React.createElement("div", {
    className: "sc-message--date",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 8,
      columnNumber: 5
    }
  }, " ", date.toLocaleTimeString([], {
    timeStyle: 'short'
  }).replace(/(\r\n|\n|\r)/gm, ""), " ");
  return /*#__PURE__*/React.createElement("a", {
    className: "sc-message--file",
    href: props.data.url,
    download: props.data.fileName,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 11,
      columnNumber: 5
    }
  }, /*#__PURE__*/React.createElement(FileIcon, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 12,
      columnNumber: 7
    }
  }), /*#__PURE__*/React.createElement("p", {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 13,
      columnNumber: 7
    }
  }, props.data.fileName), dateString);
};

export default FileMessage;
