// searchx-chat/es/components/Messages/TextMessage.js
var _this = this,
    _jsxFileName = "/Users/felipemoraes/Developer/react-team-chat-window/src/components/Messages/TextMessage.js";

import React from 'react';
import Linkify from 'react-linkify';

var TextMessage = function TextMessage(props) {
  var date = new Date(props.data.date);
  var dateString = /*#__PURE__*/React.createElement("div", {
    className: "sc-message--date",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 8,
      columnNumber: 5
    }
  }, " ", date.toLocaleTimeString([], {
    timeStyle: 'short'
  }).replace(/(\r\n|\n|\r)/gm, ""), " ");
  return /*#__PURE__*/React.createElement("div", {
    className: "sc-message--text",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 10,
      columnNumber: 10
    }
  }, /*#__PURE__*/React.createElement(Linkify, {
    properties: {
      target: '_blank'
    },
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 11,
      columnNumber: 5
    }
  }, props.data.text), dateString);
};

export default TextMessage;
