// searchx-chat/es/components/ChatWindow.js
var _jsxFileName = "/Users/felipemoraes/Developer/react-team-chat-window/src/components/ChatWindow.js";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (typeof call === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

import PropTypes from 'prop-types';
import React, { Component } from 'react';
import MessageList from './MessageList';
import UserInput from './UserInput';
import Header from './Header';

var ChatWindow = /*#__PURE__*/function (_Component) {
  _inheritsLoose(ChatWindow, _Component);

  var _super = _createSuper(ChatWindow);

  function ChatWindow(props) {
    return _Component.call(this, props) || this;
  }

  var _proto = ChatWindow.prototype;

  _proto.onUserInputSubmit = function onUserInputSubmit(message) {
    this.props.onUserInputSubmit(message);
  };

  _proto.onFilesSelected = function onFilesSelected(filesList) {
    this.props.onFilesSelected(filesList);
  };

  _proto.render = function render() {
    var messageList = this.props.messageList || [];
    var classList = ['sc-chat-window', this.props.isOpen ? 'opened' : 'closed'];
    return /*#__PURE__*/React.createElement("div", {
      className: classList.join(' '),
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 28,
        columnNumber: 7
      }
    }, /*#__PURE__*/React.createElement(Header, {
      teamName: this.props.agentProfile.teamName,
      imageUrl: this.props.agentProfile.imageUrl,
      onClose: this.props.onClose,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 29,
        columnNumber: 9
      }
    }), /*#__PURE__*/React.createElement(MessageList, {
      messages: messageList,
      imageUrl: this.props.agentProfile.imageUrl,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 34,
        columnNumber: 9
      }
    }), /*#__PURE__*/React.createElement(UserInput, {
      onSubmit: this.onUserInputSubmit.bind(this),
      onFilesSelected: this.onFilesSelected.bind(this),
      showEmoji: this.props.showEmoji,
      showFile: this.props.showFile,
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 38,
        columnNumber: 9
      }
    }));
  };

  return ChatWindow;
}(Component);

ChatWindow.propTypes = process.env.NODE_ENV !== "production" ? {
  agentProfile: PropTypes.object.isRequired,
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  onFilesSelected: PropTypes.func,
  onUserInputSubmit: PropTypes.func.isRequired,
  showEmoji: PropTypes.bool,
  showFile: PropTypes.bool
} : {};
export default ChatWindow;
