// searchx-chat/es/components/MessageList.js
var _jsxFileName = "/Users/felipemoraes/Developer/react-team-chat-window/src/components/MessageList.js";

function _createSuper(Derived) { return function () { var Super = _getPrototypeOf(Derived), result; if (_isNativeReflectConstruct()) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (typeof call === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

import React, { Component } from 'react';
import Message from './Messages';

var MessageList = /*#__PURE__*/function (_Component) {
  _inheritsLoose(MessageList, _Component);

  var _super = _createSuper(MessageList);

  function MessageList() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = MessageList.prototype;

  _proto.componentDidUpdate = function componentDidUpdate(_prevProps, _prevState) {
    this.scrollList.scrollTop = this.scrollList.scrollHeight;
  };

  _proto.render = function render() {
    var _this = this;

    return /*#__PURE__*/React.createElement("div", {
      className: "sc-message-list",
      ref: function ref(el) {
        return _this.scrollList = el;
      },
      __self: this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 12,
        columnNumber: 7
      }
    }, this.props.messages.map(function (message, i) {
      return /*#__PURE__*/React.createElement(Message, {
        message: message,
        key: i,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 14,
          columnNumber: 18
        }
      });
    }));
  };

  return MessageList;
}(Component);

export default MessageList;
